### ISSUE TITLE
[bitnami/keycloak] password authentication failed for user "bn_keycloak"

### ISSUE THREAD

**Issue by Tortoaster** [bitnami/charts#29602]
### Name and Version

bitnami/keycloak 22.2.6

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Use the configuration below
2. Create `admin-secret.yaml` with `password`
3. Create `db-secret.yaml` with `password` and `postgres-password`
4. Install the chart

I tried various other configurations as well, including:

1. Leaving out the ingress
5. Specifying the database passwords directly rather than using `existingSecret`
6. Not specifying any database passwords and having it generate random passwords
7. Not using `proxy: "edge"` and specifying `tls`, or disabling `production`
8. Specifying an `externalDatabase` instead
9. Running on ARM64 architecture

### Are you using any custom parameters or values?

```yaml
namespaceOverride: "keycloak"
production: true
resourcesPreset: "small"
proxy: "edge"
ingress:
  enabled: true
  hostname: "my.domain.com"
  tls: true
auth:
  adminUser: "admin"
  existingSecret: "admin-secret"
  passwordSecretKey: "password"
postgresql:
  auth:
    existingSecret: "db-secret"
```

### What is the expected behavior?

It authenticates successfully to the database.

### What do you see instead?

```
2024-09-25 15:25:22,819 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
2024-09-25 15:25:22,820 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to obtain JDBC connection
2024-09-25 15:25:22,821 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: FATAL: password authentication failed for user "bn_keycloak"
2024-09-25 15:25:22,824 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

### Additional information

Deployed on a k3s cluster using Rancher Fleet. I deleted the pvc and pv between each install, to make sure it's not using an old password. The database itself appears to be working as intended (it expects the right credentials), yet KeyCloak fails to authenticate, even though it can reach the database.

**Comment by javsalgar** [comment]
Hi!

Are you able to access postgresql using a regular psql client? Just to ensure that the credentials are properly set

**Comment by Tortoaster** [comment]
```shell
kubectl -n keycloak get secrets db-secret --template={{.data.password}} | base64 -d
psql -h 10.43.7.73 -U bn_keycloak
```

Just verified with these commands, the respective passwords for `bn_keycloak` and `postgres` both work!

**Comment by dgomezleon** [comment]
Hi @Tortoaster ,

Could it be related to how the secrets were created? I have created them this way and I could not reproduce the issue:

```
kubectl create secret generic admin-secret --from-literal=password=bitnami
kubectl create secret generic db-secret --from-literal=postgres-password=bitnami-pg --from-literal=password=bitnami-pass
```

**Comment by Tortoaster** [comment]
Thank you for your reply! I tried those exact commands and they worked perfectly. After some more experiments, I found that the passwords themselves were the problem. They contained characters like `$!#&%^*`. I'm not sure which of them were the culprit, but alphanumeric passwords work fine.

It is interesting that the database seemed to work with the previous passwords, but KeyCloak didn't, but that is probably an issue in KeyCloak itself rather than this chart.
